docs/modules/servers/pages/local/self-hosting.notes

```diff
@@ -0,0 +1,90 @@
+(which is slightly external to James as DNS setup is required)
+
+
+
+A "how to" regarding this is a MUST have in my opinion http://james.apache.org/howTo/index.html
+
+
+
+Just a thought based on the ideas you wrote here...
+
+But what if the mission of James were to "help people self-host their email" as an alternative to providers like gmail etc.
+
+Right now James is very obscure and is only for total tekkie geeks. If we modify the target, perhaps this could be the mission.
+
+The way of accomplishing this is:
+
+Provide in-depth "concept" articles
+Provide more "how tos" for important things
+Make James easier to install and operate
+wdyt?
+
+
+It would be more like:
+
+Apache James give you the ability to "self-host" email. Many companies and individuals are unable to develop their own mail infrastructure, but often feel trapped into using services like gmail, yahoo, etc. James provides a low-cost alternative for those who desire the freedom to use email the way they want.
+
+Or something like that.
+
+
+
+Apache James give you the ability to "self-host" email. Many companies and individuals are unable to develop their own mail infrastructure, but often feel trapped into using services like gmail, yahoo, etc. James provides a low-cost alternative for those who desire the freedom to use email the way they want.
+
+This should be the goal of the Basic server.
+
+Regarding "Advanced server":
+
+Free email servers are generaly old, written in complex languages (C) and extending their behavior is not one of there main concerns. 
+
+Apache James Advanced server have been designed with extensibility in mind. It provides several extension mechanisms allowing you to easily configure and extend the behavior of your email server in order to make it fits your needs. Java and other JVM languages can be used to write your own extensions.
+And for the Apache James Distributed server:
+
+Scaling out IMAP mail server is notoriously difficult, as Mail server generally relies on technologies that don't scale well. Apache James server had been designed on top of modern distributed middlewares that allow you to scale it out just like any other modern web applications, reducing the management overhead of large scale deployments.
+We could be thinking of, regarding the testing server:
+
+Embed a tiny, zero external dependency, easy to configure, feature complete, testing mail server to test the email logics for your code. Could be run directly in your JVM tests or spawn via docker.
+Would that go in your direction?
+
+
+For the Basic server we should manage to be less "teckie oriented".
+
+However "Advanced servers" branding should be compelling for "teckies" and discourage other operators.
+
+
+Very nice!
+
+That gives me an idea for the naming, then:
+
+Basic Server (for self-hosting - should be very simple)
+Extendable Server (for customizing - can be more complex)
+Advanced Server (for just about everything else, including "Distributed" - no holds barred)
```

Comment:
I do think the "branding" should be close (enough) to the current structure of the project.

JPA based servers, and the distributed one have very distinct feature-set, thus should be distinct. Hence I do think we should document them separately.

Regarding Extendable/Advanced, I would go with just "Advanced" in an effort to reduce the count of server to expose/maintain. This could be a move coming in the future though.

My 2 cents.

Reply:
Sorry, these notes are a bit outdated. They are mostly snippets that I copy/pasted from discussions on the mailing list, but have not yet addressed. I have taken note of your comments in the notes, though. 😀